src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -342,17 +369,71 @@ localCompletionsForParsedModule pm@ParsedModule{pm_parsed_source = L _ HsModule{
 
     thisModName = ppr hsmodName
 
+    --recordCompls = localRecordSnippetProducer pm thisModName
+
+findRecordCompl' :: ParsedModule -> T.Text -> TyClDecl GhcPs -> [CompItem]
+findRecordCompl' pmod mn x = case x of
+    DataDecl {tcdLName, tcdDataDefn} -> findRecordCompl pmod mn tcdLName tcdDataDefn
+    _ -> []
+
+findRecordCompl :: ParsedModule -> T.Text -> Located (IdP GhcPs) -> HsDataDefn GhcPs -> [CompItem]
+findRecordCompl pmod mn lname dd = catMaybes name_type''
+    where
+        conDecl = unLoc <$> dd_cons dd
+        h98 = catMaybes $ findH98 <$> conDecl
+        findH98 conDecl = case conDecl of
+          ConDeclH98{..} -> Just (unLoc con_name, con_args)
+          ConDeclGADT{} -> Nothing  -- TODO: Expand this out later
```

Comment:
Are you planning to look at this TODO?

Reply:
I was not planning to as part of this PR.  I can take a look and see if it is easy.  Would it be OK to do it as part of another PR?